mixer/cmd/build.go

```diff
@@ -149,13 +150,52 @@ var buildUpstreamFormatCmd = &cobra.Command{
 		if err := b.UnstageMixFromBump(); err != nil {
 			fail(err)
 		}
+
+		bumpNeeded, err := b.CheckBumpNeeded(true)
+		if err != nil {
+			fail(err)
+		}
+		// Loop through multiple bumps until we reach the latest format
+		for bumpNeeded {
+			cmdToRun = strings.Split("mixer build format-bump new", " ")
```

Comment:
Instead of copypasting the entire function, why not just wrap the entire original function in the bump check loop?  The premise is that it's already been called and the mix has already been staged by the time it gets here, so the initial condition of the bool can just be true.

I haven't verified this code whatsoever, but something like:

```go
	Run: func(cmd *cobra.Command, args []string) {
 		b, err := builder.NewFromConfig(configFile)
 		if err != nil {
 			fail(err)
 		}

 		bumpNeeded := true

 		// Loop through multiple bumps until we reach the latest format
 		for bumpNeeded {
			cmdToRun := strings.Split("mixer build format-bump new", " ")
	 		if config.UseNewConfig {
	 			cmdToRun = append(cmdToRun, "--new-config")
	 		}
	 		if err := b.RunCommandInContainer(cmdToRun); err != nil {
	 			fail(err)
	 		}
	 
	 		// Set the upstream version to the previous format's latest version
	 		b.UpstreamVerUint32 -= 10
	 		b.UpstreamVer = strconv.FormatUint(uint64(b.UpstreamVerUint32), 10)
	 		vFile := filepath.Join(b.Config.Builder.VersionPath, b.UpstreamVerFile)
	 		if err := ioutil.WriteFile(vFile, []byte(b.UpstreamVer), 0644); err != nil {
	 			fail(err)
	 		}
	 		cmdToRun = strings.Split("mixer build format-bump old", " ")
	 		if config.UseNewConfig {
	 			cmdToRun = append(cmdToRun, "--new-config")
	 		}
	 		if err := b.RunCommandInContainer(cmdToRun); err != nil {
	 			fail(err)
	 		}
	 		// Set the upstream version back to what the user originally tried to build
	 		if err := b.UnstageMixFromBump(); err != nil {
	 			fail(err)
	 		}
	 		bumpNeeded, err = b.CheckBumpNeeded(true)
			if err != nil {
				fail(err)
			}
 		}
	}
```

Reply:
This is a better place to put it for sure!